sklearn/utils/_test_common/instance_generator.py

```diff
@@ -457,6 +471,78 @@
     ),
 }
 
+# This dictionary stores parameters for specific checks. It also enables running the
+# same check with multiple instances of the same estimator with different parameters.
+# The special key "*" allows to apply the parameters to all checks.
+CHECK_PARAMS: dict = {
```

Comment:
Maybe rename this a more explicit name such as `ESTIMATOR_CHECK_CUSTOM_PARAMS` or `PER_ESTIMATOR_CHECK_PARAMS`?